sparse/_coo/core.py

```diff
@@ -621,6 +621,8 @@ def nnz(self):
         >>> np.count_nonzero(x) == s.nnz
         True
         """
+        if self.shape == ():
+            return len(self.data)
         return self.coords.shape[1]
```

Comment:
What I'd be concerned about is... Why is `self.coords.shape[1]` not correct for a scalar. Could we look into the constructor?

Reply:
Zero dimensional array also has zero dimensional coordinates at the moment:

```python
a = sparse.COO.from_numpy(np.array(1))
a.coords

#array([], shape=(0, 0), dtype=int64)
```

This makes sense to me (the length of a coordinate array in 0-dimensions would also be zero).

Reply:
The shape of coords should theoretically always be `(ndim, nnz)`, so if there is a scalar array... It should be `(0, 1)`.